src/request.ts

```diff
@@ -77,9 +77,7 @@ async function startFetchRequest(
     if (resOrErr instanceof Response) {
       if (retryOpts.statusCodes.includes(resOrErr.status)) {
         if (maxRetry === retryCount) {
-          const err = new Error('Request failed');
-          err.name = 'RequestRetryError';
-          throw err;
+          return false
```

Comment:
return false here so it stops retrying and makes the caller returns the response on L144 here:
https://github.com/jsforce/jsforce/blob/658d7f24783327e74c8d5e5b2bfa9e542d718acd/src/request.ts#L126-L144